servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -3989,10 +3989,22 @@ void validateServicePrincipal(final String memberName, final String caller) {
         // are typically domains (like for ci/cd) where services
         // are dynamic and do not need to be registered in Athenz
 
-        if (!validateServiceMemberSkipDomains.contains(domainName)) {
-            if (dbService.getServiceIdentity(domainName, serviceName, true) == null) {
-                throw ZMSUtils.requestError("Principal " + memberName + " is not a valid service", caller);
+        for (String skipDomain : validateServiceMemberSkipDomains) {
```

Comment:
validateServiceMemberSkipDomains was created as a Set to have simple contains check. if we're treating as a regular list instead, you should change the definition to be an ArrayList instead of a Set.

Reply:
thank you. I've fixed it.